projects/packages/jetpack-mu-wpcom/src/features/wpcom-themes/wpcom-theme-tracking.php

```diff
@@ -77,3 +78,68 @@ function wpcom_themes_tracks_switch_theme( $new_theme_name, $new_theme, $old_the
 	Common\wpcom_record_tracks_event( 'wpcom_theme_switch', $event_props );
 }
 add_action( 'switch_theme', 'wpcom_themes_tracks_switch_theme', 12, 3 );
+
+/**
+ * Record a theme upload via wp-admin.
+ *
+ * @param Theme_Upgrader $upgrader   Upgrader instance.
+ * @param array          $hook_extra Upgrade data.
+ * @return void
+ */
+function wpcom_themes_tracks_upload_theme( $upgrader, $hook_extra ) {
+	if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+		// Simple sites can't upload themes with wp-admin so should never reach here,
+		// but just in case, we exit early.
+		return;
+	}
+
+	if ( ! isset( $hook_extra['type'] ) || $hook_extra['type'] !== 'theme' || ! isset( $hook_extra['action'] ) ) {
+		return;
+	}
+
+	$theme_slug = $upgrader->result['destination_name'] ?? '';
+
+	if ( ! $theme_slug ) {
+		return;
+	}
+
+	$theme        = wp_get_theme( $theme_slug );
+	$theme_props  = wpcom_themes_tracks_get_theme_props( $theme );
+	$upload_props = array(
+		'source' => '', // Intentionally left empty for debugging purposes.
+		'type'   => 'upload',
+	);
+
+	// Trying to distinguish between a manual upload vs a standard install or upgrade. It's not security sensitive.
+	// phpcs:disable WordPress.Security.NonceVerification.Recommended,WordPress.Security.ValidatedSanitizedInput.MissingUnslash,WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+	$action    = $_GET['action'] ?? '';
+	$overwrite = $_GET['overwrite'] ?? '';
+	// phpcs:enable WordPress.Security.NonceVerification.Recommended,WordPress.Security.ValidatedSanitizedInput.MissingUnslash,WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+
+	if ( $hook_extra['action'] === 'install' && ! $action ) {
+		// Install theme via wp-admin UI
+		$upload_props['source'] = 'wp-admin';
+		$upload_props['type']   = 'install';
```

Comment:
What's the point of tracking a `wpcom_theme_upload` event when the user has not uploaded a theme? If we want to track more scenarios, maybe the event name should be more generic?